### PR TITLE
Update features heuristic to include more unstable and private feature name patterns

### PR DESCRIPTION
**Pull request by era**
Addresses #461, by adding `_`, `unstable-` and `unstable_` prefixes to ignore.

### CODE REVIEW

File: src/rustdoc_gen.rs [hunk 1]
```diff
@@ -143,11 +146,27 @@ impl<'a> CrateSource<'a> {
             String::from("no_std"),
         ]);
 
-        let determine = |feature_name: &String| {
-            !features_ignored_by_default.contains(feature_name) && !feature_name.starts_with("__")
+        let prefix_ignored_by_default = vec![
+            String::from("_"),
+            String::from("unstable-"),
+            String::from("unstable_"),
+        ];
```
[the remark this answers is not in the document]
Comment (era) [reply src/rustdoc_gen.rs:153]: Oh, it would! I'm cleaning it to be `&str` instead.

**Comment by era** [comment]
Thank you for the review, I properly setup my editor, so I should not have that type of problem anymore :).

Thank you again for all the tips,

**Review by obi1kenobi — APPROVED** [review]

**Comment by obi1kenobi** [comment]
@era awesome work! Are you happy staying on the CLI side for now or would you like to try your hand at writing some lints as well? It's all the same to me, so it's your call :)

The docs for writing lints are [here](https://github.com/obi1kenobi/cargo-semver-checks/blob/main/CONTRIBUTING.md#adding-a-new-lint) and I think I linked to a few easy-ish lints to start with in a prior chat.

**Comment by era** [comment]
Ah, thank you for pointing me to the lint docs, I think I will try to do one lint to see how it goes! Thank you again for all mentoring around the project!!

**Comment by obi1kenobi** [comment]
No worries, thanks for all the awesome contributions — it's my pleasure to help!

**Pull request merged**